### PR TITLE
Make toggle accessible

### PR DESCRIPTION
**Pull request by ranquild**
Epic: #19696

- Moves Toggle to core/components
- Adds keyboard interactions
- Adds focus ring
- Forwards ref
- Fixes `reflow` issue with `left` animation by replacing it with `transform`

[Storybook](https://19749-core-toggle--61d4abcd959e8c003aa51d4c.chromatic.com/?path=/story/core-checkbox--default)

### CODE REVIEW

File: frontend/src/metabase/core/components/Toggle/Toggle.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  title: "Core/Toggle",
+  component: Toggle,
+  argTypes: { onChange: { action: "onChange" } },
+};
+
+const Template: ComponentStory<typeof Toggle> = args => {
+  return <Toggle {...args} />;
+};
+
+export const Default = Template.bind({});
```
Comment (kdoh) [inline frontend/src/metabase/core/components/Toggle/Toggle.stories.tsx:15]: It'd be nice to also add use of the props like `small` or `color` here so we can document and see and understand how this component can shift in usage.

**Review by kdoh — CHANGES_REQUESTED** [review]
Overall looks like it's in the right direction just left one note about adding more examples in the `.stories` file.

File: frontend/src/metabase/core/components/Toggle/Toggle.styled.tsx [hunk 2]
```diff
@@ -2,56 +2,60 @@ import styled from "styled-components";
...
   }
 };
 
-export const ToggleRoot = styled.a<ToggleRootProps>`
+export const ToggleRoot = styled.input<ToggleRootProps>`
+  appearance: none;
   position: relative;
   display: inline-block;
   color: ${props => props.currentColor ?? color("brand")};
-  box-sizing: border-box;
```
Comment (kdoh) [inline frontend/src/metabase/core/components/Toggle/Toggle.styled.tsx:35]: Curious about this removal. Is the idea that we already define this at `*` in our global CSS, or was there another reason?

File: frontend/src/metabase/core/components/Toggle/Toggle.stories.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/core/components/Toggle/Toggle.stories.tsx:18]: They already there - just click on controls in storybook

File: frontend/src/metabase/core/components/Toggle/Toggle.styled.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/core/components/Toggle/Toggle.styled.tsx:35]: Yes, it is defined for `input`s.

**Review by kulyk — APPROVED** [review]
👏

File: frontend/src/metabase/sharing/components/EmailAttachmentPicker.unit.spec.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ describe("EmailAttachmentPicker", () => {
     it("should have a Toggle that is not toggled", () => {
       const toggle = screen.getByLabelText("Attach results");
       expect(toggle).toBeInTheDocument();
-      expect(toggle).toHaveAttribute("aria-checked", "false");
+      expect(toggle).toHaveProperty("checked", false);
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/sharing/components/EmailAttachmentPicker.unit.spec.js]: Sure, thanks!

File: frontend/src/metabase/core/components/Toggle/Toggle.styled.tsx [hunk 4]
```diff
@@ -2,56 +2,60 @@ import styled from "styled-components";
 import { color } from "metabase/lib/colors";
 
 export interface ToggleRootProps {
-  isSmall?: boolean;
-  isSelected?: boolean;
+  checked?: boolean;
+  small?: boolean;
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/core/components/Toggle/Toggle.styled.tsx:6]: The idea was not to change the existing API in any way. We could (and would) improve it later.

**Review by kdoh — APPROVED** [review]

**Pull request merged**